src/views/Map/Map.tsx

```diff
@@ -125,8 +124,14 @@ export function MapPage() {
             userModel?.hasFeature(AppFeature.EVENTS_LIST) &&
             userModel?.feeds && <EventListPanel />
           }
-          layers={userModel?.hasFeature(AppFeature.MAP_LAYERS_PANEL) && <MapLayersList />}
-          legend={userModel?.hasFeature(AppFeature.LEGEND_PANEL) && <Legend />}
+          layersAndLegends={
+            <LayersAndLegendsWidget
+              layers={layersPanel}
```

Comment:
why not just userModel?.hasFeature(AppFeature.LEGEND_PANEL) && layersPanel ? no need for isLegendOn prop then 
same with legend

Reply:
in that case you don't need src/widgets/LayersAndLegends/index.tsx